robot/EDA/resources/EDA.py

```diff
@@ -381,3 +424,187 @@ def get_namespace_prefix(self, name):
+        """ Selects Contacts tab - as passed in to the function """
+        locator_menu = eda_lex_locators["tab_menu"].format(tab)
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["tab_tab"].format(tab)
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def select_value_from_listbox(self, title, value):
+        """ Selects value from a listbox/dropdown identified by title """
+        locator_title = eda_lex_locators["listbox"]["title"].format(title)
+        locator_value = eda_lex_locators["listbox"]["value"].format(value)
+        self.selenium.click_element(locator_title)
+        time.sleep(1)
+        self.selenium.click_element(locator_value)
+
+    def select_xpath(self, loc, value):
+        """ Selects the correct xpath by checking if it exists on the page
+            from the given list of locator possibilities
+        """
+        locators = eda_lex_locators[loc].values()
+
+        for i in locators:
+            locator = self.format_all(i, value)
+            if self.check_if_element_exists(locator):
+                return locator
+
+        assert "Button with the provided locator not found"
+
+    def wait_for_locator_with_timeout(self, timeout, path, *args):
+        """Waits for specified element to be available
+            by checking every 1 seconds for given number of seconds
+        """
+        locator = self.get_eda_locator(path, *args)
+
+        i = int(timeout)/7
+        while i > 1:
+            try:
+                logger.console("\n" + "Check:" + str(round(i)) + " - Waiting for locator")
+                self.selenium.get_webelement(locator)
+                return True
+            except Exception:
+                time.sleep(1)
+                i -= 1
+
+        self.builtin.log("Timed out waiting for locator with path " + locator)
+        return False
+
+    def wait_for_new_window(self, title):
+        """ Waits for specified window to be available
+            by checking every 1 seconds for 25 times
+        """
+        window_found = False
+
+        for i in range(25):
+            i += 1
+            time.sleep(1)
+            titles = self.selenium.get_window_titles()
+            for j in titles:
+                if j == title:
+                    window_found = True
+                    return window_found
+
+            if window_found:
```

Comment:
Since you call return a couple lines above, you don't need the if statement here. You can remove this whole if statement.

Reply:
This is someone else's code, and is outside the scope of the Preferred Phone work.  Will consider in the EDA rewrite.

Reply:
I don't understand that comment. You're the one adding the code to the repository.

Reply:
True

I removed the code since I haven't found a need for it yet.  It looked like a good possibility when I was designing.

Reply:
This function has now been removed.